app/src/main/java/factorio/train/analyser/analyser/Deadlock_Analyser.java

```diff
@@ -0,0 +1,122 @@
+package factorio.train.analyser.analyser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import factorio.train.analyser.graph.Graph;
+import factorio.train.analyser.graph.Section;
+import factorio.train.analyser.graph.Node;
+import java.util.Map.Entry;
+
+public class Deadlock_Analyser {
+
+    private Graph graph;
+
+    public Deadlock_Analyser() {}
+
+    public void setGraph(Graph graph) {
+        this.graph = graph;
+    }
+
+    public ArrayList<Result> deadlockCheck() {
+        if(graph != null){
+            ArrayList<Result> result = deadlockDetection(graph.getSections());
+            return result;
+        }else{
+            return null;
+        }
+    }
+
+    private ArrayList<Result> deadlockDetection(ArrayList<Section> sections) {
```

Comment:
pls add Javadoc comments to all the methods and fields. I recommend using copilot.